src/components/SignUpForm/index.js

```diff
@@ -1,41 +1,61 @@
 import React from 'react';
-import { bool, func, string } from 'prop-types';
-import { Field, reduxForm } from 'redux-form';
-import { ActivityIndicator, Button, Text, View } from 'react-native';
+import { func } from 'prop-types';
+import { Button, View } from 'react-native';
 
 import Input from 'components/common/Input';
-import * as constraints from 'utils/constraints';
+import useForm from 'hooks/useForm';
+import useValidation from 'hooks/useValidation';
+import signUpValidations from 'validations/signUpValidations';
 import translate from 'utils/i18n';
+import ErrorView from 'components/common/ErrorView';
 import styles from './styles';
 
-const SignUpForm = ({ handleSubmit, error, submitting }) => (
-  <View onSubmit={handleSubmit}>
-    {error && <Text>{error}</Text>}
-    <Field name="email" label={translate('SIGN_UP.email')} component={Input} />
-    <Field name="password" label={translate('SIGN_UP.password')} component={Input} password />
-    <Field
-      name="passwordConfirmation"
-      label={translate('SIGN_UP.passwordConfirmation')}
-      component={Input}
-      password
-    />
-    {submitting ? (
-      <ActivityIndicator />
-    ) : (
+const FIELDS = {
+  email: 'email',
+  password: 'password',
+  passwordConfirmation: 'passwordConfirmation',
+};
+
+const SignUpForm = ({ onSubmit }) => {
+  const validator = useValidation(signUpValidations);
+  const [values, , errors, , handleValueChange, handleSubmit] = useForm(
+    {
+      initialValues: {},
+      onSubmit,
+      validator,
+    },
+    [onSubmit],
+  );
+
+  return (
+    <>
+      <Input
+        label={translate('SIGN_UP.email')}
+        value={values.email}
+        onChangeText={text => handleValueChange(FIELDS.email, text)}
```

Comment:
What do you think about providing a `inputProps` function from the `useForm` hook, so on every input you have to just do something like this:
```jsx
<Input
    label={translate('SIGN_UP.email')}
    { ...inputProps(FIELDS.email) }
/>
```

Being inputProps something like this:

```js
const inputProps = (field) => ({
    value: values[field],
    onChangeText: text => handleValueChange(field, text),
});
```

Reply:
I know this would require some workaround for other type of inputs, but I think it's doable

Reply:
It could be a good idea, but I don't think that belongs on the form Hook, since it would be a really specific cases. The idea of hooks is to encapsulate behaviour, helper functions should go separate. I'll add it as a separate hook